### DISCUSSION TITLE
get initial value and label from back end during mount?

### DISCUSSION THREAD

**Discussion by judgej** [wireui/wireui#564] [General]
The `select` component, once rendered, needs to make a call back to the server to get the display values/markup/whatever. Is there any way - just like for any other livewire form field - to populate these things in the component when it is first mounted? I am looking for how to avoid the browser callbacks when the initial page render has enough information to supply this on the first render.

**Comment by judgej** [comment]
Maybe a slot can do this? Can a single "selected" option be added to the content of the `x-select` option on rendering?

**Comment by ph7jack** [comment]
Hi @judgej 
When using the async select, you need to provide an endpoint and a value/model to be selected in the initial page load.
Then, the `select` will fetch the selected record.
Adding the selected value and the label manually will make it complex.

**Reply by judgej** [reply]
It is certainly easy to use, the way works now, no doubt about that.

What I am thinking of here is perhaps closer to the Livewire philosphy of keeping the requests back to the server to a minimum, by effectively pre-loading the page uo front. This is possibly because all Livewire content is generated by server-side views.

Now, I think the thing making it coplicated to do this with the WireUI select is that the content is generated client-side, on the browser, with *data* used to update the selected value of the select item. However, if the server had a view that produced the exact same content that the client produced for a selected item (or items), then could it conceivably provide that intial select content when the select is first mounted? I'm thinking out loud here, and realise nothing is trivial, but is this an approach that could work?

Imagine a table listing 100 items. Each item may have two or three select items to edit in-position. That currently would involve a hundred or more requests back to the server just to mount the initial page. The delay in seeing the initial selected value appear is very real. And yes, the design could be changed to aavoid putting the application into that position, but I'm not sure UI design should be led by technical limitations.

If the select component needs an array of JSON to render, then could that same array of JSON not be provided with the initial page load?

**Reply by ph7jack** [reply]
I understand your points, but I think it's not useful.
If you have 100 selects in your context, you can use the `:options="[...]"` prop instead of using it as async.
The select options HTML is rendered by the JS to prevent delay in loading a page with multiple selects, in the past, the HTML was rendered in the backend, but it increase the page load a lot.
Now, the HTML is rendered by the js on demand.
You can provide the initial "json" to be rendered, just give it to the select component as a `slot` or by the `options` prop.

**Reply by judgej** [reply]
It would be nice to be able to use both. I could supply `options` with just the single value that the item is currently selected to, and also supply an async route for fetching updates and queries when the user goes into the item.

Delivering JSON through an API is the absolute right approach, so no argument there.

Aaaand...just read your last point. Are you saying I *can* give both an async route *and* a list of one option (the one selected)? Does that one option go into the cache to avoid the callback to fetch it from the API on mounting? If so, that's perfect, just what I was looking for.

**Reply by judgej** [reply]
Ah, no, can't use both at the same time without an error.

> The {async-data} attribute cannot be used with {options} attribute.

I guess my question comes down to how easy it will be to allow both, so it's an async select, but on mounting, the back-end can bootstrap it with some lookup data that it can use initially.
